include/open62541/server.h

```diff
@@ -1285,6 +1285,114 @@ UA_Server_triggerEvent(UA_Server *server, const UA_NodeId eventNodeId, const UA_
+ * passed to make the condition unexposed.
+ * @param server The server object
+ * @param conditionType The NodeId of the node representation of the ConditionType
+ * @param conditionName The name of the condition to be created
+ * @param conditionSource The NodeId of the Condition Source (Parent of the Condition)
+ * @param hierarchialReferenceType The NodeId of Hierarchical ReferenceType between Condition and its source
+ * @param outNodeId The NodeId of the created Condition
+ * @return The StatusCode of the UA_Server_triggerEvent method */
+UA_StatusCode UA_EXPORT
+UA_Server_createCondition(UA_Server *server, const UA_NodeId conditionType,
+                          UA_QualifiedName conditionName, const UA_NodeId conditionSource,
+                          const UA_NodeId hierarchialReferenceType, UA_NodeId *outConditionId);
+
+/**
+ * set the value of condition field.
+ * @param server The server object
+ * @param condition The NodeId of the node representation of the Condition Instance
+ * @param variantValue Variant Value to be written to the Field
+ * @param type Variant Type.
+ * @param fieldName Name of the Field in which the value should be written
+ * @return The StatusCode of the UA_Server_triggerEvent method*/
+UA_StatusCode UA_EXPORT
+UA_Server_setConditionField(UA_Server *server, const UA_NodeId condition,
+                            const void* variantValue, UA_UInt16 type,
+                            const UA_QualifiedName fieldName);
+
+/**
+ * set the value of property of condition field.
+ * @param server The server object
+ * @param condition The NodeId of the node representation of the Condition Instance
+ * @param variantValue Variant Value to be written to the Field
+ * @param type Variant Type
+ * @param variableFieldName Name of the Field which has a property
+ * @param variablePropertyName Name of the Field Property in which the value should be written
+ * @return The StatusCode of the UA_Server_triggerEvent method*/
+UA_StatusCode UA_EXPORT
+UA_Server_setConditionVariableFieldProperty(UA_Server *server, const UA_NodeId condition,
+                                            const void *variantValue, UA_UInt16 type,
+                                            const UA_QualifiedName variableFieldName,
+                                            const UA_QualifiedName variablePropertyName);
+
+/**
+ * triggers an event only for an enabled condition. The condition list is updated then with the
+ * last generated EventId.
+ * @param server The server object
+ * @param condition The NodeId of the node representation of the Condition Instance
+ * @param conditionSource The NodeId of the node representation of the Condition Source
+ * @param outEventId last generated EventId
+ * @return The StatusCode of the UA_Server_triggerEvent method*/
+UA_StatusCode UA_EXPORT
+UA_Server_triggerConditionEvent(UA_Server *server, const UA_NodeId condition,
```

Comment:
Why can't we just use the normal `UA_Server_triggerEvent` with the respective `eventNodeId` and `originId`?

Reply:
The last generated EventIds should be stored in an intern list so we could manage the Conditions. (EvenIds are needed for most Condition method calls)